core-bundle/src/Doctrine/Backup/Dumper.php

```diff
@@ -0,0 +1,141 @@
+    }
+
+    public function doDump(Connection $connection, CreateConfig $config): \Generator
+    {
+        yield 'SET FOREIGN_KEY_CHECKS = 0;';
+        $schemaManager = $connection->createSchemaManager();
+        $platform = $connection->getDatabasePlatform();
+
+        foreach ($this->getTablesToDump($schemaManager, $config) as $table) {
+            yield from $this->dumpSchema($platform, $table);
+            yield from $this->dumpData($connection, $table);
+        }
+
+        yield from $this->dumpViews($schemaManager, $platform);
+
+        // Triggers are currently not supported (contributions welcome!)
+
+        yield 'SET FOREIGN_KEY_CHECKS = 1;';
+    }
+
+    private function dumpViews(AbstractSchemaManager $schemaManager, AbstractPlatform $platform): \Generator
+    {
+        foreach ($schemaManager->listViews() as $view) {
+            yield sprintf('-- BEGIN VIEW %s', $view->getName());
+            yield $platform->getCreateViewSQL($view->getQuotedName($platform), $view->getSql()).';';
+        }
+    }
+
+    private function dumpSchema(AbstractPlatform $platform, Table $table): \Generator
+    {
+        yield sprintf('-- BEGIN STRUCTURE %s', $table->getName());
+        yield sprintf('DROP TABLE IF EXISTS `%s`;', $table->getName());
+
+        foreach ($platform->getCreateTableSQL($table) as $statement) {
+            yield $statement.';';
+        }
+    }
+
+    private function dumpData(Connection $connection, Table $table): \Generator
+    {
+        yield sprintf('-- BEGIN DATA %s', $table->getName());
+        $values = [];
+
+        foreach ($table->getColumns() as $column) {
+            if ($column->getType() instanceof BinaryType) {
+                $values[] = sprintf('HEX(`%s`) AS `%s`', $column->getName(), $column->getName());
+            } else {
+                $values[] = sprintf('`%s` AS `%s`', $column->getName(), $column->getName());
+            }
+        }
+
+        $rows = $connection->executeQuery(sprintf('SELECT %s FROM `%s`', implode(', ', $values), $table->getName()));
+
+        foreach ($rows->fetchAllAssociative() as $row) {
+            $insertColumns = [];
+            $insertValues = [];
+
+            foreach ($row as $columnName => $value) {
+                $insertColumns[] = sprintf('`%s`', $columnName);
+                $insertValues[] = $this->formatValueForDump($value, $table->getColumn($columnName), $connection);
+            }
+
+            yield sprintf(
+                'INSERT INTO `%s` (%s) VALUES (%s);',
+                $table->getName(),
+                implode(', ', $insertColumns),
+                implode(', ', $insertValues)
+            );
+        }
+    }
+
+    private function formatValueForDump($value, Column $column, Connection $connection): string
+    {
+        if (null === $value) {
+            return 'NULL';
+        }
+
+        if ($column->getType() instanceof BinaryType) {
+            return sprintf('UNHEX(%s)', $connection->quote($value));
```

Comment:
`mysqldump` converts binary fields to `0xd82c13f00ce411ec9433d8bbc1145c99` for example with the `--hex-blob` option instead of `UNHEX('D82C13F00CE411EC9433D8BBC1145C99')`. May be we should do the same here to, in order to avoid any issues with binary data (e.g. https://github.com/contao/contao/pull/3631#issuecomment-975400899).